c2rust-transpile/src/translator/mod.rs

```diff
@@ -3085,13 +3093,8 @@ impl<'c> Translation<'c> {
         let ty = self.convert_type(type_id)?;
         let mut zero = mk().lit_expr(mk().int_unsuffixed_lit(0));
         if is_static && !pointee.qualifiers.is_const {
-            let mut qtype = pointee;
-            qtype.qualifiers.is_const = true;
-            let ty_ = self
-                .type_converter
-                .borrow_mut()
-                .convert_pointer(&self.ast_context, qtype)?;
-            zero = mk().cast_expr(zero, ty_);
+            let ty_ = self.convert_pointee_type(pointee.ctype)?;
+            zero = mk().cast_expr(zero, mk().ptr_ty(ty_));
```

Comment:
Should this still be translating the pointee type with `is_const` set to `true`?

(moved from [other PR](https://github.com/immunant/c2rust/pull/1446#discussion_r2582671135))

Reply:
Ah, that makes sense. Thanks for the explanation.